src/app/components/sing-in/sing-in.component.ts

```diff
@@ -12,13 +13,14 @@ import { SingInService } from '../../service';
 export class SingInComponent {
   email?: User;
 
-  constructor(private singInService: SingInService,
+  constructor(private router: Router, private singInService: SingInService,
     private dialogRef: DialogRef,
     @Inject(DIALOG_DATA) public data: any
   ) { }
 
   onBtnClick() {
     this.singInService.login(this.email);
+    this.router.navigate([this.data]);
```

Comment:
Is there any case when this.data is not set?

Reply:
Fixed